tools/lib/tests/test_caching.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+
+import unittest
+from tools.lib.url_file import URLFile
+
+class TestFileDownload(unittest.TestCase):
+
+  def test_downloads(self):
+    small_file_url = "https://raw.githubusercontent.com/commaai/openpilot/master/SAFETY.md"
```

Comment:
ideally the test shouldn't need internet. you can make a temp file, fill it with junk bytes, and refer to it like this `file:///path/to/the/tmp/file`

Reply:
The tests also need to check the partial download functionality using range headers. If we want to run locally we need something like flask. However we already ran into some issues with azure implementing the range headers in a non standard way.

Reply:
Yeah not sure there is any benefit from not needing internet, there are some bugs in previous version that are imo not reproducible in local testing, for example getting the whole file if the file is large enough and the end of range is not specified

Reply:
can we change it to a qlog file to make the egress bandwidth as small as possible?
also azure storage account emulator (that runs in docker): https://github.com/Azure/Azurite